core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedScheduleGenerator.scala

```diff
@@ -601,16 +589,31 @@ class CostBasedScheduleGenerator(
   }
 
   /**
-    * The cost function used by the search. Takes a region plan, generates one or more (to be done in the future)
-    * schedules based on the region plan, and calculates the cost of the schedule(s) using Cost Estimator. Uses the cost
-    * of the best schedule (currently only considers one schedule) as the cost of the region plan.
+    * Takes a region DAG, generates one or more (to be done in the future) schedules based on the region DAG, allocates
+    * resources to each region in the region DAG, and calculates the cost of the schedule(s) using Cost Estimator. Uses
+    * the cost of the best schedule (currently only considers one schedule) as the cost of the region DAG.
     *
     * @return A cost determined by the cost estimator.
     */
-  private def evaluate(regionPlan: RegionPlan): Double = {
+  private def allocateResourcesAndEvaluateCost(
+      regionDAG: DirectedAcyclicGraph[Region, RegionLink]
```

Comment:
what's the difference between a `regionDAG` and a `regionPlan`?

Reply:
`regionDAG` is a jgrapht DAG and is used internally by the search process and needs to be updated here so that the final search result includes the resourceConfig.
`regionPlan` is a wrapper class we created for representing regionDAG as the result of the search process and is currently only used as the input for generating a schedule.

Reply:
can we use region plan for this input argument? we can always build the regionDAG inside the cost estimator